Add malloc_conf_2_conf_harder

davidtgoldblatt opened the pull request:
This comes in handy when you're just a user of a canary system who wants to change settings set by the configuration system itself (and so you control the binary but not the environment variables, which can override your settings).

The silly name is deliberate -- it's a hint that this shouldn't really be present in "serious" code.

yinan1048576 commented:
This is a handy feature. I had to manually overwrite the options before.

I also wonder: how do you deal with the jemalloc tests? What I was doing was: if I had to manually overwrite some option, some tests may fail which would block the canary runs, so I had to manually surround my manual overwrite with something like `ifndef JEMALLOC_JET`.

Implementation wise, I think you'd better put a fifth element in `opts_explain` (in `malloc_conf_init_helper()`) as well as in `opts_cache` (in `malloc_conf_init()`).

davidtgoldblatt commented:
Updated per comments. Changed the name to `malloc_conf_2_conf_harder`, which both better captures the intent and also sounds more like the [namesake](https://itunes.apple.com/us/movie/die-hard-2-die-harder/id578539479).
